### DISCUSSION TITLE
How to enable TX Fee paid by miner?

### DISCUSSION THREAD

**Discussion by ei8ht187** [coinfoundry/miningcore#1065] [Q&A]
Hello there, 
in the meantime I got the PoC installation running and mining on Ropsten works without any problems. Except for the payout module. 
We want that the TX fees are paid by the miners, so the sending amount is brutto. 
How can this be achieved? The Option "keepTransactionFees" is set to true, but also with false there is no difference. Is this even possible with Miningcore? 
The whole config.json looks as follows: 

> {
    "$schema": "https://raw.githubusercontent.com/oliverw/miningcore/master/src/Miningcore/config.schema.json",
    "logging": {
        "level": "info",
        "enableConsoleLog": true,
        "enableConsoleColors": true,
        "logFile": "",
        "apiLogFile": "",
        "logBaseDirectory": "",
        "perPoolLogFile": true
    },
    "banning": {
        "manager": "Integrated",
        "banOnJunkReceive": false,
        "banOnInvalidShares": false
    },
    "notifications": {
        "enabled": false,
        "email": {
            "host": "smtp.example.com",
            "port": 587,
            "user": "user",
            "password": "password",
            "fromAddress": "info@yourpool.org",
            "fromName": "pool support"
        },
        "admin": {
            "enabled": false,
            "emailAddress": "user@example.com",
            "notifyBlockFound": true
        }
    },
    "persistence": {
        "postgres": {
            "host": "timescale",
            "port": 5432,
            "user": "miningcore",
            "password": "[SNIP]",
            "database": "miningcore"
        }
    },
    "paymentProcessing": {
        "enabled": true,
        "interval": 100,
        "shareRecoveryFile": "recovered-shares.txt"
    },
    "api": {
        "enabled": true,
        "listenAddress": "127.0.0.1",
        "port": 4000,
        "metricsIpWhitelist": [],
        "rateLimiting": {
            "disabled": true,
            "rules": [
                {
                    "Endpoint": "127.0.0.1",
                    "Period": "1s",
                    "Limit": 5
                }
            ],
            "ipWhitelist": [
                ""
            ]
        }
    },
    "clusterName": "eu1",
    "pools": [
      {
        "id": "eth1",
        "enabled": true,
        "coin": "ethereum",
        "address": "0x68[SNIP]FB",
        "rewardRecipients": [
          {
            "type": "op",
            "address": "0x49[SNIP]Bc",
            "percentage": 1.0
          }
        ],
        "blockRefreshInterval": 120,
        "clientConnectionTimeout": 600,
        "banning": {
          "enabled": true,
          "time": 600,
          "invalidPercent": 50,
          "checkThreshold": 50
        },
        "ports": {
          "8888": {
            "name": "GPU-SMALL",
            "listenAddress": "*",
            "difficulty": 1,
            "varDiff": {
              "minDiff": 1,
              "maxDiff": null,
              "targetTime": 15,
              "retargetTime": 90,
              "variancePercent": 30
            }
          }
        },
        "chainTypeOverride": "Ethereum",
        "daemons": [
          {
            "host": "192.168.100.4",
            "port": 8545,
            "user": "",
            "password": ""
          }
        ],
        "paymentProcessing": {
            "enabled": true,
            "minimumPayment": 1,
            "payoutScheme": "PPLNS",
            "payoutSchemeConfig": {
                "factor": 0.5
            },
            "gas": 21000,
            "maxFeePerGas": 50000000000,
            "keepUncles": false,
            "keepTransactionFees": true
        }
      }
    ]
}

Kind regards

ei8ht

**Comment by oliverw** [comment]
Did the blocks you test with actual have transactions?

**Reply by ei8ht187** [reply]
Just our own, more or less. 
Can run a script to gernerate TXs and GAS fee for testing.
However, when I check the TX, I receive exactly the amount thats visible in the logs, but the amount for the whole TX is the amount visible in the logs + gas fee.
![Screenshot 2022-01-04 20-33-41](https://user-images.githubusercontent.com/72598876/148113784-51cd4d55-d6c1-4457-b216-417debefd03f.png)

**Reply by oliverw** [reply]
Hmm, maybe I'm tired but I can't find the problem. The logic is pretty simple:

https://github.com/oliverw/miningcore/blob/master/src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs#L135

**Reply by ei8ht187** [reply]
The problem we have is: 
Miner gets paid 5.9325 ETH. That's the value that is visible in the MC logs (
> miningcore_1  | [2022-01-04 19:30:00.5390] [I] [eth1] [Ethereum Payout Handler] Sending 5.9325 ETH to 0x12[SNIP]e2 
> miningcore_1  | [2022-01-04 19:30:00.5913] [I] [eth1] [Ethereum Payout Handler] Payment transaction id: 0x9d[SNIP]35 

When you check the total of this transaction is 5.93252351. It is higher than the amount of ETH the miner gets, meaning, the pool is paying the transaction fee. 
Maybe this is a problem caused by empty blocks. Will generate some non empty blocks and check again.

When it will be helpful for you to have walletaddresses, logs and/or TX ID I'm happy to give them to you directly via mail or similar :).
For the moment, I'll try to understand some C# code :)

**Comment by James1847** [comment]
I want to confirm KeepTransactionFees == false means the tx fee is going to paid by miner, alright?
@oliverw
